### PR TITLE
feat(IoTDA): add a  resource to manage device linkage rule

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
add a  resource to manage device linkage rule

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/iotda' TESTARGS='-run=TestAccDeviceLinkageRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run=TestAccDeviceLinkageRule_basic -timeout 360m -parallel 4
=== RUN   TestAccDeviceLinkageRule_basic
=== PAUSE TestAccDeviceLinkageRule_basic
=== CONT  TestAccDeviceLinkageRule_basic
--- PASS: TestAccDeviceLinkageRule_basic (88.83s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     89.458s
```

### CODE REVIEW

File: docs/resources/iotda_device_linkage_rule.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+
+* `actions` - (Required, List) Specifies the list of the actions, at most 10 actions.
+The [actions](#IoTDA_actions) structure is documented below.
+
+* `trigger_logic` - (Optional, String) Specifies the logical relationship between multiple triggers.
+The options are as follows:
+  + **and**: All of the triggers is met.
+  + **or**: Any of the triggers is met.
+
+Defaults to `and`.
```
Comment (Lance52259) [inline docs/resources/iotda_device_linkage_rule.md]: wrong format, missing indent (two space).

File: docs/resources/iotda_device_linkage_rule.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID in UUID format.
+
+## Import
+
+device linkage rules can be imported using the `id`, e.g.
```
Comment (Lance52259) [inline docs/resources/iotda_device_linkage_rule.md]: The first letter of the sentence needs to be capitalized.

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go [hunk 3]
```diff
@@ -0,0 +1,816 @@
...
+	}
+	log.Printf("[DEBUG] Create IoTDA device linkage rule params: %#v", rule)
+
+	resp, err := client.CreateRule(&model.CreateRuleRequest{Body: rule})
+	if err != nil {
+		return diag.Errorf("error creating IoTDA device linkage rule: %s", err)
+	}
+
+	if resp.RuleId == nil {
+		return diag.Errorf("error creating IoTDA device linkage rule: id is not found in API response")
```
Comment (Lance52259) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go]: s/id/ID

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go [hunk 4]
```diff
@@ -0,0 +1,816 @@
...
+
+			"space_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"triggers": {
+				Type:     schema.TypeSet,
+				Optional: true,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go]: it's required in the doc

File: docs/resources/iotda_device_linkage_rule.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+
+* `triggers` - (Required, List) Specifies the list of the triggers, at most 10 triggers.
+The [triggers](#IoTDA_triggers) structure is documented below.
+
+* `actions` - (Required, List) Specifies the list of the actions, at most 10 actions.
+The [actions](#IoTDA_actions) structure is documented below.
+
+* `trigger_logic` - (Optional, String) Specifies the logical relationship between multiple triggers.
+The options are as follows:
+  + **and**: All of the triggers is met.
```
Comment (Jason-Zhang9309) [inline docs/resources/iotda_device_linkage_rule.md]: s/is/are

File: docs/resources/iotda_device_linkage_rule.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+
+* `value` - (Required, String) Specifies the Rvalue of a data comparison expression. When the `operator` is `between`,
+the Rvalue represents the minimum and maximum values, separated by commas, such as "20,30",
+which means greater than or equal to 20 and less than 30.
+
+* `trigger_strategy` - (Optional, String) Specifies the trigger strategy. The options are as follows:
+  + **pulse**: When the data reported by the device meets the conditions, the rule can be triggered.
+  + **reverse**: Repetition suppression. For example, if an alarm is configured to be triggered when the battery level
+   is lower than 20%, the alarm will be triggered once the battery initially drops below 20% but will not be triggered
+   again each time the battery drops to a lower level.
```
Comment (Jason-Zhang9309) [inline docs/resources/iotda_device_linkage_rule.md:157]: add a description of default value

File: docs/resources/iotda_device_linkage_rule.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+* `data_validatiy_period` - (Optional, String) Specifies data validity period. For example, if Data Validity Period is
```
Comment (Jason-Zhang9309) [inline docs/resources/iotda_device_linkage_rule.md]: it's int in the schema

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go [hunk 8]
```diff
@@ -0,0 +1,816 @@
...
+	c := meta.(*config.Config)
+	region := c.GetRegion(d)
+	client, err := c.HcIoTdaV5Client(region)
+	if err != nil {
+		return diag.Errorf("error creating IoTDA v5 client: %s", err)
+	}
+
+	rule, err := buildDeviceLinkageRuleParams(d)
+	if err != nil {
+		return diag.FromErr(err)
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go:362]: should add a error message here, there is no error message in func buildDeviceLinkageRuleParams

File: docs/resources/iotda_device_linkage_rule.md (hunk 5 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md]: fix

File: docs/resources/iotda_device_linkage_rule.md (hunk 6 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md:157]: ok

File: docs/resources/iotda_device_linkage_rule.md (hunk 7 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md]: fix

File: docs/resources/iotda_device_linkage_rule.md (hunk 1 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md]: ok

File: docs/resources/iotda_device_linkage_rule.md (hunk 2 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md]: fix

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go (hunk 3 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go]: ok

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go (hunk 4 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go]: fix

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go (hunk 8 above)
Comment (hmbig2) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_device_linkage_rule.go:362]: actually , there are some error messages in func

File: docs/resources/iotda_device_linkage_rule.md [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+      - **method**: optional, the command name defined in the product model.
+      - **body**: optional, the message body of the command, which contains key-value pairs, each key is the parameter
+        name of the command in the product model. The specific format requires application and device conventions.
+
+<a name="IoTDA_smn_forwarding"></a>
+The `smn_forwarding` block supports:
+
+* `region` - (Required, String) Specifies the region to which the SMN belongs.
+
+* `project_id` - (Required, String) Specifies the project ID to which the SMN belongs.
```
Comment (niuzhenguo) [inline docs/resources/iotda_device_linkage_rule.md]: can we make the project default to region as what we does in provider configuration

File: huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_linkage_rule_test.go [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+  }
+
+  effective_period {
+    start_time   = "00:00"
+    end_time     = "23:59"
+    days_of_week = "1,2,3,4,5,6"
+  }
+
+  depends_on = [
+    huaweicloud_iotda_device.test,
```
Comment (niuzhenguo) [inline huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_linkage_rule_test.go:250]: do we need this depends_on? seems you already do reference to it at L238

Comment (hmbig2) [reply huaweicloud/services/acceptance/iotda/resource_huaweicloud_iotda_device_linkage_rule_test.go:250]: There are 3 test cases,  prevent the device from being deleted

File: docs/resources/iotda_device_linkage_rule.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/iotda_device_linkage_rule.md]: OK， change it to optional

**Pull request merged**